### DISCUSSION TITLE
When I use terminal vi or vim to open a file, the half-screen display is not available after exiting

### DISCUSSION THREAD

**Discussion by saowu** [sourcelair/xterm.js#4624] [Q&A] [answered]
When I use terminal vi or vim to open a file, the half-screen display is not available after exiting
anticipate result：
![image](https://user-images.githubusercontent.com/33569619/224206019-18e3edcf-4586-4756-bf44-3939f6f827e7.png)

Problem effect：
![image](https://user-images.githubusercontent.com/33569619/224205967-bb797329-ea20-4758-b438-886c5b24e270.png)

## Details
- Browser and browser version: Google Chrome 110.0.5481.177（正式版本） (arm64)
- OS version: MacOS 12.6.3
- xterm.js version: 
  -   "xterm": "^5.1.0",
  -  "xterm-addon-attach": "^0.8.0",
  -  "xterm-addon-fit": "^0.7.0"
  -  "vue": "3.2.31"
  - 

### Steps to reproduce

1. Connection terminal
2. Open any file using vi/vim
3. Exit vi/vim

**Comment by dmarkowski** [comment]
I am not sure if it's the same problem but when I open a file by 'less' or 'vi' it uses only half of the available space.  After exiting them terminal returns to full screen. Do you know why 'less' or 'vi' are using only half of the screen? I've got "xterm": "^5.1.0", "xterm-addon-fit": "^0.7.0" on react 17 project.

**Comment by jerch** [comment]
@dmarkowski Whats the size terminal size in cols/rows (`term.cols` and `term.rows` in console.log)? Whats the output of `tput lines` and `tput cols`?

**Comment by dmarkowski** [comment]
@jerch when I add a listener to `term.onResize` it prints me `{cols: 165, rows: 38}`. My server doesn't have `tput` command but when I use `stty size` it prints me `24 80`

**Comment by jerch** [comment, marked as answer]
Ah well, you need to apply the size changes to the PTY as well (values should stay in sync).

**Comment by dmarkowski** [comment]
@jerch thx for help. After setting PTY size it works. I thought that 'xterm-addon-fit' will do this out of the box but I was wrong.

**Comment by jerch** [comment]
It cannot do that, as it does not know, how to inform the backend of the changes (this highly depends on your backend integration). The fit addon only updates the terminal metrics from div container size changes.
